src/components/FilterDatastreamData.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2024 SECO Mind Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { AstarteInterfaceValues } from 'astarte-client';
+import React, { useEffect, useState } from 'react';
+import { Button, Form, Col, Row, Spinner } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+interface FilterDatastreamDataProps {
+  data: AstarteInterfaceValues | null;
+  fetchData?: (path?: string, since?: string, to?: string) => void;
+  requestingData?: boolean;
+}
+
+const FilterDatastreamData = ({
+  data,
+  fetchData,
+  requestingData,
+}: FilterDatastreamDataProps): React.ReactElement => {
+  const [fromTime, setFromTime] = useState<Date | null>(null);
+  const [toTime, setToTime] = useState<Date | null>(null);
+  const [path, setPath] = useState<string>('');
+  const [paths, setPaths] = useState<string[]>([]);
+
+  const extractPaths = (data: AstarteInterfaceValues): string[] => {
+    const result: string[] = [];
+
+    if (typeof data === 'object' && data !== null && !Array.isArray(data)) {
+      const firstKey = Object.keys(data)[0];
+      if (firstKey) {
+        const firstValue = data[firstKey];
+        if (typeof firstValue === 'object' && firstValue !== null && !Array.isArray(firstValue)) {
+          const subKeys = Object.keys(firstValue);
+
+          if (subKeys.length === 1) {
+            const subKey = subKeys[0];
+            result.push(`/${firstKey}/${subKey}`);
+          } else if (subKeys.length > 1) {
+            result.push(`/${firstKey}`);
+          }
+        } else {
+          result.push(`/${firstKey}`);
+        }
+      }
+    }
+
+    return result;
+  };
```

Comment:
1. What is the purpose of this function? What does it do? The code is not immediately clear.
2. Does it compute the existing paths where interface data were published? Maybe we could improve the function's name.